From 52864ecc6f0e20635125b1c6812c8ea3c1b0ce48 Mon Sep 17 00:00:00 2001
From: Jason Hsu <Jason-Hsu@quantatw.com>
Date: Wed, 10 Sep 2025 18:13:00 +0800
Subject: [PATCH] ARM: dts: aspeed: ventura: add missing cable presence gpios
 v2

Since the 9 x cable presence GPIOs for 4 x AALC cables, 5 x Rack level
leak cables are missing from the Ventura DTS, we added those to the
device tree.

This commit additionally adds IRQ support in GPIO expanders under i2c10 to
handle edge-triggered, and reorders the nodes based on upstream suggestions.

LORE link:
https://lore.kernel.org/all/20250903120600.632041-1-pkleequanta@gmail.com/T/#u

Tested:
The physical port mapping and line-names have been verified to math
correctly.

We have updated patch 0007 v2(version 2) to fix the issue where Bus3
i2c3mux0ch6 and i2c3mux0ch7 should not be part of the i2c3mux0ch5 configuration.
We have adjusted their positions accordingly.

Signed-off-by: Jason Hsu <Jason-Hsu@quantatw.com>
---
 .../aspeed/aspeed-bmc-facebook-ventura.dts    | 615 ++++++++++--------
 1 file changed, 346 insertions(+), 269 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
index 54528a560..72076cf2f 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-ventura.dts
@@ -53,8 +53,8 @@ chosen {
 	iio-hwmon {
 		compatible = "iio-hwmon";
 		io-channels = <&adc0 0>, <&adc0 1>, <&adc0 2>, <&adc0 3>,
-			      <&adc0 4>, <&adc0 5>, <&adc0 6>, <&adc0 7>,
-			      <&adc1 2>;
+			<&adc0 4>, <&adc0 5>, <&adc0 6>, <&adc0 7>,
+			<&adc1 2>;
 	};
 
 	leds {
@@ -624,7 +624,7 @@ memory@80000000 {
 		reg = <0x80000000 0x80000000>;
 	};
 
-    p1v8_bmc_aux: regulator-p1v8-bmc-aux {
+	p1v8_bmc_aux: regulator-p1v8-bmc-aux {
 		compatible = "regulator-fixed";
 		regulator-name = "p1v8_bmc_aux";
 		regulator-min-microvolt = <1800000>;
@@ -640,7 +640,7 @@ p2v5_bmc_aux: regulator-p2v5-bmc-aux {
 		regulator-always-on;
 	};
 
-    spi1_gpio: spi {
+	spi1_gpio: spi {
 		compatible = "spi-gpio";
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -659,6 +659,50 @@ tpm@0 {
 	};
 };
 
+&fmc {
+	status = "okay";
+	flash@0 {
+		status = "okay";
+		m25p,fast-read;
+		label = "bmc";
+		spi-max-frequency = <50000000>;
+#include "openbmc-flash-layout-128.dtsi"
+	};
+	flash@1 {
+		status = "okay";
+		m25p,fast-read;
+		label = "alt-bmc";
+		spi-max-frequency = <50000000>;
+	};
+};
+
+&mdio0 {
+	status = "okay";
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rmii";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii4_default>;
+	fixed-link {
+		speed = <100>;
+		full-duplex;
+	};
+};
+
+&ehci0 {
+	status = "okay";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&uhci {
+	status = "okay";
+};
+
 &adc0 {
 	vref-supply = <&p1v8_bmc_aux>;
 	status = "okay";
@@ -668,7 +712,6 @@ &adc0 {
 		&pinctrl_adc2_default &pinctrl_adc3_default
 		&pinctrl_adc4_default &pinctrl_adc5_default
 		&pinctrl_adc6_default &pinctrl_adc7_default>;
-
 };
 
 &adc1 {
@@ -679,29 +722,198 @@ &adc1 {
 	pinctrl-0 = <&pinctrl_adc10_default>;
 };
 
-&ehci0 {
-	status = "okay";
+&gpio0 {
+	gpio-line-names =
+	/*A0-A7*/	"","","","","","","","",
+	/*B0-B7*/	"BATTERY_DETECT","","","BMC_READY","","","","",
+	/*C0-C7*/	"","","","","","","","",
+	/*D0-D7*/	"","","","","","","","",
+	/*E0-E7*/	"","","","","","","","",
+	/*F0-F7*/	"","","","","","","","",
+	/*G0-G7*/	"","","","","","","","",
+	/*H0-H7*/	"","","","","","","","",
+	/*I0-I7*/	"","","","","","","","",
+	/*J0-J7*/	"","","","","","","","",
+	/*K0-K7*/	"","","","","","","","",
+	/*L0-L7*/	"","","","","","","","",
+	/*M0-M7*/	"","","","","","","","",
+	/*N0-N7*/	"","","","","","","","",
+	/*O0-O7*/	"","","","","","","","USBDBG_IPMI_EN_L",
+	/*P0-P7*/	"","","","","","","","",
+	/*Q0-Q7*/	"","","","","","FM_MDIO_SW_SEL","","",
+	/*R0-R7*/	"","","","","","","","",
+	/*S0-S7*/	"","","","","","","","",
+	/*T0-T7*/	"","","","","","","","",
+	/*U0-U7*/	"","","","","","","","",
+	/*V0-V7*/	"","","","","","","","",
+	/*W0-W7*/	"","","","","","","","",
+	/*X0-X7*/	"","","","","","","","",
+	/*Y0-Y7*/	"","","","","","","","",
+	/*Z0-Z7*/	"","","","","","","","";
 };
 
-&ehci1 {
+&sgpiom0 {
 	status = "okay";
+	ngpios = <128>;
+	bus-frequency = <2000000>;
+
+	gpio-line-names =
+	/*"input pin","output pin"*/
+	/*A0 - A7*/
+	"power-chassis-good","power-chassis-control",
+	"host0-ready","WATER_VALVE_CLOSED_N",
+	"wPRSNT_RETURN_PLD_R_N","FM_MDIO_SW_SEL_PLD",
+	"wPRSNT_SUPPLY_PLD_R_N","FM_88E6393X_BIN_UPDATE_EN_N",
+	"LEAK3_DETECT","",
+	"LEAK4_DETECT","",
+	"RETURN_CNTL_FB_D_R","",
+	"SUPPLY_CNTL_FB_D_R","",
+	/*B0 - B7*/
+	"IOEXP0_INT_N","",
+	"IOEXP11_INT_N","",
+	"IOEXP10_INT_N","",
+	"IOEXP9_INT_N","",
+	"RPU_2_READY_SPARE_PLD_R","",
+	"RPU_2_READY_PLD_R","",
+	"RPU_READY_SPARE_PLD_R","",
+	"RPU_READY_PLD_R","",
+	/*C0 - C7*/
+	"wAALC_RPU_READY","",
+	"LEAK0_DETECT","",
+	"LEAK1_DETECT","",
+	"LEAK2_DETECT","",
+	"PRSNT_COMPUTE_TRAY1_N","",
+	"PRSNT_COMPUTE_TRAY2_N","",
+	"PRSNT_COMPUTE_TRAY3_N","",
+	"PRSNT_COMPUTE_TRAY4_N","",
+	/*D0 - D7*/
+	"PRSNT_COMPUTE_TRAY5_N","",
+	"PRSNT_COMPUTE_TRAY6_N","",
+	"PRSNT_COMPUTE_TRAY7_N","",
+	"PRSNT_COMPUTE_TRAY8_N","",
+	"PRSNT_NVS_TRAY1_N","",
+	"PRSNT_NVS_TRAY2_N","",
+	"PRSNT_COMPUTE_TRAY11_N","",
+	"PRSNT_COMPUTE_TRAY12_N","",
+	/*E0 - E7*/
+	"PRSNT_COMPUTE_TRAY13_N","",
+	"PRSNT_COMPUTE_TRAY14_N","",
+	"PRSNT_COMPUTE_TRAY15_N","",
+	"PRSNT_COMPUTE_TRAY16_N","",
+	"PRSNT_COMPUTE_TRAY17_N","",
+	"PRSNT_COMPUTE_TRAY18_N","",
+	"PRSNT_NVS_TRAY3_N","",
+	"PRSNT_NVS_TRAY4_N","",
+	/*F0 - F7*/
+	"PRSNT_NVS_TRAY5_N","",
+	"PRSNT_NVS_TRAY6_N","",
+	"PRSNT_NVS_TRAY7_N","",
+	"PRSNT_NVS_TRAY8_N","",
+	"PRSNT_NVS_TRAY9_N","",
+	"PRSNT_COMPUTE_TRAY9_N","",
+	"PRSNT_COMPUTE_TRAY10_N","",
+	"SMALL_LEAK_COMPUTE_TRAY1_N","",
+	/*G0 - G7*/
+	"SMALL_LEAK_COMPUTE_TRAY2_N","",
+	"SMALL_LEAK_COMPUTE_TRAY3_N","",
+	"SMALL_LEAK_COMPUTE_TRAY4_N","",
+	"SMALL_LEAK_COMPUTE_TRAY5_N","",
+	"SMALL_LEAK_COMPUTE_TRAY6_N","",
+	"SMALL_LEAK_COMPUTE_TRAY7_N","",
+	"SMALL_LEAK_COMPUTE_TRAY8_N","",
+	"SMALL_LEAK_NVS_TRAY1_N","",
+	/*H0 - H7*/
+	"SMALL_LEAK_NVS_TRAY2_N","",
+	"SMALL_LEAK_COMPUTE_TRAY11_N","",
+	"SMALL_LEAK_COMPUTE_TRAY12_N","",
+	"SMALL_LEAK_COMPUTE_TRAY13_N","",
+	"SMALL_LEAK_COMPUTE_TRAY14_N","",
+	"SMALL_LEAK_COMPUTE_TRAY15_N","",
+	"SMALL_LEAK_COMPUTE_TRAY16_N","",
+	"SMALL_LEAK_COMPUTE_TRAY17_N","",
+	/*I0 - I7*/
+	"SMALL_LEAK_COMPUTE_TRAY18_N","",
+	"SMALL_LEAK_NVS_TRAY3_N","",
+	"SMALL_LEAK_NVS_TRAY4_N","",
+	"SMALL_LEAK_NVS_TRAY5_N","",
+	"SMALL_LEAK_NVS_TRAY6_N","",
+	"SMALL_LEAK_NVS_TRAY7_N","",
+	"SMALL_LEAK_NVS_TRAY8_N","",
+	"SMALL_LEAK_NVS_TRAY9_N","",
+	/*J0 - J7*/
+	"SMALL_LEAK_COMPUTE_TRAY9_N","",
+	"SMALL_LEAK_COMPUTE_TRAY10_N","",
+	"PWRGD_COMPUTE_TRAY1_N","",
+	"PWRGD_COMPUTE_TRAY2_N","",
+	"PWRGD_COMPUTE_TRAY3_N","",
+	"PWRGD_COMPUTE_TRAY4_N","",
+	"PWRGD_COMPUTE_TRAY5_N","",
+	"PWRGD_COMPUTE_TRAY6_N","",
+	/*K0 - K7*/
+	"PWRGD_COMPUTE_TRAY7_N","",
+	"PWRGD_COMPUTE_TRAY8_N","",
+	"PWRGD_NVS_TRAY1_PWROK_N","",
+	"PWRGD_NVS_TRAY2_PWROK_N","",
+	"PWRGD_COMPUTE_TRAY11_N","",
+	"PWRGD_COMPUTE_TRAY12_N","",
+	"PWRGD_COMPUTE_TRAY13_N","",
+	"PWRGD_COMPUTE_TRAY14_N","",
+	/*L0 - L7*/
+	"PWRGD_COMPUTE_TRAY15_N","",
+	"PWRGD_COMPUTE_TRAY16_N","",
+	"PWRGD_COMPUTE_TRAY17_N","",
+	"PWRGD_COMPUTE_TRAY18_N","",
+	"PWRGD_NVS_TRAY3_PWROK_N","",
+	"PWRGD_NVS_TRAY4_PWROK_N","",
+	"PWRGD_NVS_TRAY5_PWROK_N","",
+	"PWRGD_NVS_TRAY6_PWROK_N","",
+	/*M0 - M7*/
+	"PWRGD_NVS_TRAY7_PWROK_N","",
+	"PWRGD_NVS_TRAY8_PWROK_N","",
+	"PWRGD_NVS_TRAY9_PWROK_N","",
+	"PWRGD_COMPUTE_TRAY9_N","",
+	"PWRGD_COMPUTE_TRAY10_N","",
+	"LEAK_DETECT_COMPUTE_TRAY1_N","",
+	"LEAK_DETECT_COMPUTE_TRAY2_N","",
+	"LEAK_DETECT_COMPUTE_TRAY3_N","",
+	/*N0 - N7*/
+	"LEAK_DETECT_COMPUTE_TRAY4_N","",
+	"LEAK_DETECT_COMPUTE_TRAY5_N","",
+	"LEAK_DETECT_COMPUTE_TRAY6_N","",
+	"LEAK_DETECT_COMPUTE_TRAY7_N","",
+	"LEAK_DETECT_COMPUTE_TRAY8_N","",
+	"LEAK_DETECT_NVS_TRAY1_N","",
+	"LEAK_DETECT_NVS_TRAY2_N","",
+	"LEAK_DETECT_COMPUTE_TRAY11_N","",
+	/*O0 - O7*/
+	"LEAK_DETECT_COMPUTE_TRAY12_N","",
+	"LEAK_DETECT_COMPUTE_TRAY13_N","",
+	"LEAK_DETECT_COMPUTE_TRAY14_N","",
+	"LEAK_DETECT_COMPUTE_TRAY15_N","",
+	"LEAK_DETECT_COMPUTE_TRAY16_N","",
+	"LEAK_DETECT_COMPUTE_TRAY17_N","",
+	"LEAK_DETECT_COMPUTE_TRAY18_N","",
+	"LEAK_DETECT_NVS_TRAY3_N","",
+	/*P0 - P7*/
+	"LEAK_DETECT_NVS_TRAY4_N","",
+	"LEAK_DETECT_NVS_TRAY5_N","",
+	"LEAK_DETECT_NVS_TRAY6_N","",
+	"LEAK_DETECT_NVS_TRAY7_N","",
+	"LEAK_DETECT_NVS_TRAY8_N","",
+	"LEAK_DETECT_NVS_TRAY9_N","",
+	"LEAK_DETECT_COMPUTE_TRAY9_N","",
+	"LEAK_DETECT_COMPUTE_TRAY10_N","";
 };
 
-&fmc {
+&wdt1 {
 	status = "okay";
-	flash@0 {
-		status = "okay";
-		m25p,fast-read;
-		label = "bmc";
-		spi-max-frequency = <50000000>;
-#include "openbmc-flash-layout-128.dtsi"
-	};
-	flash@1 {
-		status = "okay";
-		m25p,fast-read;
-		label = "alt-bmc";
-		spi-max-frequency = <50000000>;
-	};
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdtrst1_default>;
+	aspeed,reset-type = "soc";
+	aspeed,external-signal;
+	aspeed,ext-push-pull;
+	aspeed,ext-active-high;
+	aspeed,ext-pulse-duration = <256>;
 };
 
 &i2c0 {
@@ -940,10 +1152,10 @@ i2c3mux0ch3: i2c@3 {
 			#size-cells = <0>;
 			reg = <3>;
 
-			// Fan Board 0 FRU
-			eeprom@56 {
-				compatible = "atmel,24c128";
-				reg = <0x56>;
+			adc@1f {
+				compatible = "ti,adc128d818";
+				reg = <0x1f>;
+				ti,mode = /bits/ 8 <1>;
 			};
 
 			fan_leds_g1_gpio: gpio@21 {
@@ -963,16 +1175,16 @@ fan_leds_g1_gpio: gpio@21 {
 				"", "";
 			};
 
-			adc@1f {
-				compatible = "ti,adc128d818";
-				reg = <0x1f>;
-				ti,mode = /bits/ 8 <1>;
-			};
-
 			adc@35 {
 				compatible = "maxim,max11617";
 				reg = <0x35>;
 			};
+
+			// Fan Board 0 FRU
+			eeprom@56 {
+				compatible = "atmel,24c128";
+				reg = <0x56>;
+			};
 		};
 
 		i2c3mux0ch4: i2c@4 {
@@ -980,10 +1192,10 @@ i2c3mux0ch4: i2c@4 {
 			#size-cells = <0>;
 			reg = <4>;
 
-			// Fan Board 1 FRU
-			eeprom@56 {
-				compatible = "atmel,24c128";
-				reg = <0x56>;
+			adc@1f {
+				compatible = "ti,adc128d818";
+				reg = <0x1f>;
+				ti,mode = /bits/ 8 <1>;
 			};
 
 			fan_leds_g2_gpio: gpio@21 {
@@ -1003,16 +1215,16 @@ fan_leds_g2_gpio: gpio@21 {
 				"", "";
 			};
 
-			adc@1f {
-				compatible = "ti,adc128d818";
-				reg = <0x1f>;
-				ti,mode = /bits/ 8 <1>;
-			};
-
 			adc@35 {
 				compatible = "maxim,max11617";
 				reg = <0x35>;
 			};
+
+			// Fan Board 1 FRU
+			eeprom@56 {
+				compatible = "atmel,24c128";
+				reg = <0x56>;
+			};
 		};
 
 		i2c3mux0ch5: i2c@5 {
@@ -1026,12 +1238,12 @@ pwm@20 {
 				#address-cells = <1>;
 				#size-cells = <0>;
 				channel@2 {
-					    reg = <2>;
-						sensor-type = "TACH";
+					reg = <2>;
+					sensor-type = "TACH";
 				};
 				channel@5 {
-					    reg = <5>;
-						sensor-type = "TACH";
+					reg = <5>;
+					sensor-type = "TACH";
 				};
 			};
 
@@ -1094,6 +1306,7 @@ i2c3mux0ch6: i2c@6 {
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <6>;
+
 			// REAR-IO Board FRU
 			eeprom@56 {
 				compatible = "atmel,24c128";
@@ -1117,17 +1330,19 @@ &i2c4 {
 &i2c5 {
 	status = "okay";
 
-    // VR TEMP U399
+	// VR TEMP U399
 	temperature-sensor@4c {
 		compatible = "ti,tmp75";
 		reg = <0x4c>;
 	};
-    // VR TEMP U397
+
+	// VR TEMP U397
 	temperature-sensor@4d {
 		compatible = "ti,tmp75";
 		reg = <0x4d>;
 	};
-    // BRICK TEMP U398
+
+	// BRICK TEMP U398
 	temperature-sensor@4e {
 		compatible = "ti,tmp75";
 		reg = <0x4e>;
@@ -1137,7 +1352,8 @@ temperature-sensor@4f {
 		compatible = "ti,tmp75";
 		reg = <0x4f>;
 	};
-    // RMC FRU
+
+	// RMC FRU
 	eeprom@54 {
 		compatible = "atmel,24c128";
 		reg = <0x54>;
@@ -1191,19 +1407,19 @@ ipmb@10 {
 &i2c9 {
 	status = "okay";
 
-    // SCM TEMP SENSOR
+	// SCM TEMP SENSOR
 	temperature-sensor@4b {
 		compatible = "ti,tmp75";
 		reg = <0x4b>;
 	};
 
-    // SCM FRU EEPROM
+	// SCM FRU EEPROM
 	eeprom@50 {
 		compatible = "atmel,24c128";
 		reg = <0x50>;
 	};
 
-    // BSM FRU EEPROM
+	// BSM FRU EEPROM
 	eeprom@56 {
 		compatible = "atmel,24c64";
 		reg = <0x56>;
@@ -1213,11 +1429,87 @@ eeprom@56 {
 &i2c10 {
 	status = "okay";
 
+	gpio@10 {
+		compatible = "nxp,pca9555";
+		reg = <0x10>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&sgpiom0>;
+		interrupts = <16 IRQ_TYPE_LEVEL_LOW>;
+
+		gpio-line-names =
+		"IT_STOP_PUMP_SPARE_2", "IT_STOP_PUMP_2",
+		"IT_STOP_PUMP_SPARE", "IT_STOP_PUMP",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "";
+	};
+
 	power-monitor@14 {
 		compatible = "infineon,xdp710";
 		reg = <0x14>;
 	};
 
+	gpio@19 {
+		compatible = "nxp,pca9555";
+		reg = <0x19>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&sgpiom0>;
+		interrupts = <22 IRQ_TYPE_LEVEL_LOW>;
+
+		gpio-line-names =
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"wIT_GEAR_RPU_2_LINK_PRSNT_SPARE_N_R", "wIT_GEAR_RPU_2_LINK_PRSNT_N_R",
+		"wIT_GEAR_RPU_LINK_PRSNT_SPARE_N_R", "wIT_GEAR_RPU_LINK_PRSNT_N_R",
+		"", "",
+		"", "";
+	};
+
+	gpio@1a {
+		compatible = "nxp,pca9555";
+		reg = <0x1a>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&sgpiom0>;
+		interrupts = <20 IRQ_TYPE_LEVEL_LOW>;
+
+		gpio-line-names =
+		"wPRSNT_LEAK1_SENSOR_R_PLD_N", "wPRSNT_LEAK0_SENSOR_R_PLD_N",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "wPRSNT_LEAK2_SENSOR_R_PLD_N";
+	};
+
+	gpio@1b {
+		compatible = "nxp,pca9555";
+		reg = <0x1b>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-parent = <&sgpiom0>;
+		interrupts = <18 IRQ_TYPE_LEVEL_LOW>;
+
+		gpio-line-names =
+		"", "",
+		"", "",
+		"", "wPRSNT_LEAK4_SENSOR_R_PLD_N",
+		"wPRSNT_LEAK3_SENSOR_R_PLD_N", "",
+		"", "",
+		"", "",
+		"", "",
+		"", "";
+	};
+
 	power-monitor@44 {
 		compatible = "lltc,ltc4286";
 		reg = <0x44>;
@@ -1278,224 +1570,9 @@ tray_leds_g6_gpio: gpio@26 {
 		#gpio-cells = <2>;
 	};
 
-    // LED Board FRU
+	// LED Board FRU
 	eeprom@56 {
 		compatible = "atmel,24c128";
 		reg = <0x56>;
 	};
 };
-
-&mdio0 {
-	status = "okay";
-};
-
-&mac3 {
-	status = "okay";
-	phy-mode = "rmii";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_rmii4_default>;
-	fixed-link {
-		speed = <100>;
-		full-duplex;
-	};
-};
-
-&uhci {
-	status = "okay";
-};
-
-&gpio0 {
-	gpio-line-names =
-	/*A0-A7*/	"","","","","","","","",
-	/*B0-B7*/	"BATTERY_DETECT","","","BMC_READY","","","","",
-	/*C0-C7*/	"","","","","","","","",
-	/*D0-D7*/	"","","","","","","","",
-	/*E0-E7*/	"","","","","","","","",
-	/*F0-F7*/	"","","","","","","","",
-	/*G0-G7*/	"","","","","","","","",
-	/*H0-H7*/	"","","","","","","","",
-	/*I0-I7*/	"","","","","","","","",
-	/*J0-J7*/	"","","","","","","","",
-	/*K0-K7*/	"","","","","","","","",
-	/*L0-L7*/	"","","","","","","","",
-	/*M0-M7*/	"","","","","","","","",
-	/*N0-N7*/	"","","","","","","","",
-	/*O0-O7*/	"","","","","","","","USBDBG_IPMI_EN_L",
-	/*P0-P7*/	"","","","","","","","",
-	/*Q0-Q7*/	"","","","","","FM_MDIO_SW_SEL","","",
-	/*R0-R7*/	"","","","","","","","",
-	/*S0-S7*/	"","","","","","","","",
-	/*T0-T7*/	"","","","","","","","",
-	/*U0-U7*/	"","","","","","","","",
-	/*V0-V7*/	"","","","","","","","",
-	/*W0-W7*/	"","","","","","","","",
-	/*X0-X7*/	"","","","","","","","",
-	/*Y0-Y7*/	"","","","","","","","",
-	/*Z0-Z7*/	"","","","","","","","";
-};
-
-&sgpiom0 {
-	status = "okay";
-	ngpios = <128>;
-	bus-frequency = <2000000>;
-};
-
-&sgpiom0 {
-	gpio-line-names =
-    /*"input pin","output pin"*/
-	/*A0 - A7*/
-	"power-chassis-good","power-chassis-control",
-	"host0-ready","WATER_VALVE_CLOSED_N",
-	"wPRSNT_RETURN_PLD_R_N","FM_MDIO_SW_SEL_PLD",
-	"wPRSNT_SUPPLY_PLD_R_N","FM_88E6393X_BIN_UPDATE_EN_N",
-	"LEAK3_DETECT","",
-	"LEAK4_DETECT","",
-	"RETURN_CNTL_FB_D_R","",
-	"SUPPLY_CNTL_FB_D_R","",
-    /*B0 - B7*/
-	"IT_STOP_PUMP_SPARE_2","",
-	"IT_STOP_PUMP_2","",
-	"IT_STOP_PUMP_SPARE","",
-	"IT_STOP_PUMP","",
-	"RPU_2_READY_SPARE_PLD_R","",
-	"RPU_2_READY_PLD_R","",
-	"RPU_READY_SPARE_PLD_R","",
-	"RPU_READY_PLD_R","",
-    /*C0 - C7*/
-	"wAALC_RPU_READY","",
-	"LEAK0_DETECT","",
-	"LEAK1_DETECT","",
-	"LEAK2_DETECT","",
-	"PRSNT_COMPUTE_TRAY1_N","",
-	"PRSNT_COMPUTE_TRAY2_N","",
-	"PRSNT_COMPUTE_TRAY3_N","",
-	"PRSNT_COMPUTE_TRAY4_N","",
-	/*D0 - D7*/
-	"PRSNT_COMPUTE_TRAY5_N","",
-	"PRSNT_COMPUTE_TRAY6_N","",
-	"PRSNT_COMPUTE_TRAY7_N","",
-	"PRSNT_COMPUTE_TRAY8_N","",
-	"PRSNT_NVS_TRAY1_N","",
-	"PRSNT_NVS_TRAY2_N","",
-	"PRSNT_COMPUTE_TRAY11_N","",
-	"PRSNT_COMPUTE_TRAY12_N","",
-	/*E0 - E7*/
-	"PRSNT_COMPUTE_TRAY13_N","",
-	"PRSNT_COMPUTE_TRAY14_N","",
-	"PRSNT_COMPUTE_TRAY15_N","",
-	"PRSNT_COMPUTE_TRAY16_N","",
-	"PRSNT_COMPUTE_TRAY17_N","",
-	"PRSNT_COMPUTE_TRAY18_N","",
-	"PRSNT_NVS_TRAY3_N","",
-	"PRSNT_NVS_TRAY4_N","",
-	/*F0 - F7*/
-	"PRSNT_NVS_TRAY5_N","",
-	"PRSNT_NVS_TRAY6_N","",
-	"PRSNT_NVS_TRAY7_N","",
-	"PRSNT_NVS_TRAY8_N","",
-	"PRSNT_NVS_TRAY9_N","",
-	"PRSNT_COMPUTE_TRAY9_N","",
-	"PRSNT_COMPUTE_TRAY10_N","",
-	"SMALL_LEAK_COMPUTE_TRAY1_N","",
-	/*G0 - G7*/
-	"SMALL_LEAK_COMPUTE_TRAY2_N","",
-	"SMALL_LEAK_COMPUTE_TRAY3_N","",
-	"SMALL_LEAK_COMPUTE_TRAY4_N","",
-	"SMALL_LEAK_COMPUTE_TRAY5_N","",
-	"SMALL_LEAK_COMPUTE_TRAY6_N","",
-	"SMALL_LEAK_COMPUTE_TRAY7_N","",
-	"SMALL_LEAK_COMPUTE_TRAY8_N","",
-	"SMALL_LEAK_NVS_TRAY1_N","",
-	/*H0 - H7*/
-	"SMALL_LEAK_NVS_TRAY2_N","",
-	"SMALL_LEAK_COMPUTE_TRAY11_N","",
-	"SMALL_LEAK_COMPUTE_TRAY12_N","",
-	"SMALL_LEAK_COMPUTE_TRAY13_N","",
-	"SMALL_LEAK_COMPUTE_TRAY14_N","",
-	"SMALL_LEAK_COMPUTE_TRAY15_N","",
-	"SMALL_LEAK_COMPUTE_TRAY16_N","",
-	"SMALL_LEAK_COMPUTE_TRAY17_N","",
-	/*I0 - I7*/
-	"SMALL_LEAK_COMPUTE_TRAY18_N","",
-	"SMALL_LEAK_NVS_TRAY3_N","",
-	"SMALL_LEAK_NVS_TRAY4_N","",
-	"SMALL_LEAK_NVS_TRAY5_N","",
-	"SMALL_LEAK_NVS_TRAY6_N","",
-	"SMALL_LEAK_NVS_TRAY7_N","",
-	"SMALL_LEAK_NVS_TRAY8_N","",
-	"SMALL_LEAK_NVS_TRAY9_N","",
-	/*J0 - J7*/
-	"SMALL_LEAK_COMPUTE_TRAY9_N","",
-	"SMALL_LEAK_COMPUTE_TRAY10_N","",
-	"PWRGD_COMPUTE_TRAY1_N","",
-	"PWRGD_COMPUTE_TRAY2_N","",
-	"PWRGD_COMPUTE_TRAY3_N","",
-	"PWRGD_COMPUTE_TRAY4_N","",
-	"PWRGD_COMPUTE_TRAY5_N","",
-	"PWRGD_COMPUTE_TRAY6_N","",
-	/*K0 - K7*/
-	"PWRGD_COMPUTE_TRAY7_N","",
-	"PWRGD_COMPUTE_TRAY8_N","",
-	"PWRGD_NVS_TRAY1_PWROK_N","",
-	"PWRGD_NVS_TRAY2_PWROK_N","",
-	"PWRGD_COMPUTE_TRAY11_N","",
-	"PWRGD_COMPUTE_TRAY12_N","",
-	"PWRGD_COMPUTE_TRAY13_N","",
-	"PWRGD_COMPUTE_TRAY14_N","",
-	/*L0 - L7*/
-	"PWRGD_COMPUTE_TRAY15_N","",
-	"PWRGD_COMPUTE_TRAY16_N","",
-	"PWRGD_COMPUTE_TRAY17_N","",
-	"PWRGD_COMPUTE_TRAY18_N","",
-	"PWRGD_NVS_TRAY3_PWROK_N","",
-	"PWRGD_NVS_TRAY4_PWROK_N","",
-	"PWRGD_NVS_TRAY5_PWROK_N","",
-	"PWRGD_NVS_TRAY6_PWROK_N","",
-	/*M0 - M7*/
-	"PWRGD_NVS_TRAY7_PWROK_N","",
-	"PWRGD_NVS_TRAY8_PWROK_N","",
-	"PWRGD_NVS_TRAY9_PWROK_N","",
-	"PWRGD_COMPUTE_TRAY9_N","",
-	"PWRGD_COMPUTE_TRAY10_N","",
-	"LEAK_DETECT_COMPUTE_TRAY1_N","",
-	"LEAK_DETECT_COMPUTE_TRAY2_N","",
-	"LEAK_DETECT_COMPUTE_TRAY3_N","",
-	/*N0 - N7*/
-	"LEAK_DETECT_COMPUTE_TRAY4_N","",
-	"LEAK_DETECT_COMPUTE_TRAY5_N","",
-	"LEAK_DETECT_COMPUTE_TRAY6_N","",
-	"LEAK_DETECT_COMPUTE_TRAY7_N","",
-	"LEAK_DETECT_COMPUTE_TRAY8_N","",
-	"LEAK_DETECT_NVS_TRAY1_N","",
-	"LEAK_DETECT_NVS_TRAY2_N","",
-	"LEAK_DETECT_COMPUTE_TRAY11_N","",
-	/*O0 - O7*/
-	"LEAK_DETECT_COMPUTE_TRAY12_N","",
-	"LEAK_DETECT_COMPUTE_TRAY13_N","",
-	"LEAK_DETECT_COMPUTE_TRAY14_N","",
-	"LEAK_DETECT_COMPUTE_TRAY15_N","",
-	"LEAK_DETECT_COMPUTE_TRAY16_N","",
-	"LEAK_DETECT_COMPUTE_TRAY17_N","",
-	"LEAK_DETECT_COMPUTE_TRAY18_N","",
-	"LEAK_DETECT_NVS_TRAY3_N","",
-	/*P0 - P7*/
-	"LEAK_DETECT_NVS_TRAY4_N","",
-	"LEAK_DETECT_NVS_TRAY5_N","",
-	"LEAK_DETECT_NVS_TRAY6_N","",
-	"LEAK_DETECT_NVS_TRAY7_N","",
-	"LEAK_DETECT_NVS_TRAY8_N","",
-	"LEAK_DETECT_NVS_TRAY9_N","",
-	"LEAK_DETECT_COMPUTE_TRAY9_N","",
-	"LEAK_DETECT_COMPUTE_TRAY10_N","";
-};
-
-&wdt1 {
-	status = "okay";
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_wdtrst1_default>;
-	aspeed,reset-type = "soc";
-	aspeed,external-signal;
-	aspeed,ext-push-pull;
-	aspeed,ext-active-high;
-	aspeed,ext-pulse-duration = <256>;
-};
-- 
2.34.1

